master/internal/config/config.go

```diff
@@ -236,67 +236,42 @@ type ObservabilityConfig struct {
 	EnablePrometheus bool `json:"enable_prometheus"`
 }
 
-func readPreemptionFromScheduler(conf *resourcemanagers.SchedulerConfig) *bool {
-	if conf == nil || conf.Priority == nil {
-		return nil
-	}
-	return &conf.Priority.Preemption
-}
-
 func readPriorityFromScheduler(conf *resourcemanagers.SchedulerConfig) *int {
 	if conf == nil || conf.Priority == nil {
 		return nil
 	}
 	return conf.Priority.DefaultPriority
 }
 
-// ReadPreemptionStatus resolves the desired preemption status for a job.
-func ReadPreemptionStatus(rpName string, jobConf interface{}) bool {
+// ReadRMPreemptionStatus resolves the preemption status for a resource manager.
+func ReadRMPreemptionStatus(rpName string) bool {
 	config := GetMasterConfig()
-	var jobPreemptible bool
-	switch jobConf.(type) {
-	case *expconf.ExperimentConfig:
-		jobPreemptible = true
-	case *model.CommandConfig:
-		jobPreemptible = false
-	default:
-		panic("unexpected jobConf type")
-	}
-
-	RMPremption := false // Whether the RM supports preemption
+	return readRMPreemptionStatus(config, rpName)
+}
 
+func readRMPreemptionStatus(config *Config, rpName string) bool {
 	for _, rpConfig := range config.ResourcePools {
 		if rpConfig.PoolName != rpName {
 			continue
 		}
-		if preemption := readPreemptionFromScheduler(rpConfig.Scheduler); preemption != nil {
-			RMPremption = *preemption
-			return jobPreemptible && RMPremption
-		}
-		if rpConfig.Provider != nil && rpConfig.Provider.GCP != nil {
-			RMPremption = rpConfig.Provider.GCP.InstanceType.Preemptible
-			return jobPreemptible && RMPremption
+		if rpConfig.Scheduler != nil {
+			return rpConfig.Scheduler.GetPreemption()
 		}
 		break
 	}
 
 	// if not found, fall back to resource manager config
-	if config.ResourceManager.AgentRM != nil {
-		if preemption := readPreemptionFromScheduler(
-			config.ResourceManager.AgentRM.Scheduler,
-		); preemption != nil {
-			RMPremption = *preemption
-			return jobPreemptible && RMPremption
-		}
-	}
-
-	if config.ResourceManager.KubernetesRM != nil {
-		if config.ResourceManager.KubernetesRM.DefaultScheduler == "preemption" {
-			RMPremption = true
+	switch {
+	case config.ResourceManager.AgentRM != nil:
+		if config.ResourceManager.AgentRM.Scheduler == nil {
```

Comment:
https://github.com/determined-ai/determined/blob/95999f97ed3428c1315b3b621784fad4afc84768/master/internal/config/config.go#L152-L154 this should be impossible because of this